src/lib/optimizer/strategy/projection_combination_rule.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class ProjectionNode;
+
+/**
+ * This optimizer rule combines consecutive ProjectionNodes if possible.
```

Comment:
Could you expand on what "combines" means? What happens to the columns?

Reply:
The rule takes consecutive ProjectionNodes if the all have exactly one child/parent. The first node can have multiple parents, the last one multiple children. It removes all but the last ProjectionNode from the tree. If the last ProjectionNode contains any LQPColumnReferences to a column in one of the ProjectionNodes to be removed, then the LQPColumnReference is replaced by the actual column.

Reply:
I should have been clearer. "Could you expand the comment to say what combine means". ;)

Please add what you have just written to the file.

Reply:
Brain-lag on my part, sorry :sweat_smile: Updated the comment.